src/Author.php

```diff
@@ -33,13 +28,26 @@ class Author extends AbstractEndpoint {
 	 * @return array The array with the data of the endpoint
 	 */
 	public function endpoint_callback( \WP_REST_Request $request ) {
-		$user_id = $request->get_param( 'id' );
-		$response = [];
-		$user = get_user_by( 'id', $user_id );
+		$params = $request->get_params();
+
+		$id = ( false === $params['id'] ) ? false : absint( $params['id'] );
+
+		$slug = ( false === $params['slug'] ) ? false : trim( $params['slug'], '/' );
+
+		if ( false === $id && false === $slug ) {
+			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
```

Comment:
Well anything that removes:

- Conditional
- Switchs

Makes thing simpler for instance the code 

if ( a )  {
 x
} else {
 y
}


Creates 2 posible paths and by having 2 of that we hare having 4 posible paths.

By creating a default scenario on and having 2 paths decreases complexity by 50%. Also the code can be simplied by passing the absint, trim  and sanitize_text_field to the sanitize_callback callback. And that will make you to get ride 2 things from the endpoint_callback leaving you only the if. 

As I mentioned before is my point of view to code and write code and create solutions. Another example for this type of scenarios are search algorithms there are many of them that just works but are not the right fit to be used under normal scenarios..

> On Apr 27, 2016, at 3:41 PM, Caridad Zapatero Contreras <notifications@github.com> wrote:
> 
> In src/Author.php <https://github.com/moxie-leean/wp-endpoints-author/pull/6#discussion_r61332003>:
> 
> > @@ -33,13 +28,26 @@ class Author extends AbstractEndpoint {
> >  	 * @return array The array with the data of the endpoint
> >  	 */
> >  	public function endpoint_callback( \WP_REST_Request $request ) {
> > -		$user_id = $request->get_param( 'id' );
> > -		$response = [];
> > -		$user = get_user_by( 'id', $user_id );
> > +		$params = $request->get_params();
> > +
> > +		$id = ( false === $params['id'] ) ? false : absint( $params['id'] );
> > +
> > +		$slug = ( false === $params['slug'] ) ? false : trim( $params['slug'], '/' );
> > +
> > +		if ( false === $id && false === $slug ) {
> > +			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
> I guess its personal preference, there is not much I can say about that.
> 
> However, I understand that we will receive only 1 parameter, so one of these 2 lines is always going to throw a warning:
> 
> $id =  absint( $params['id'] );
> $slug = trim( sanitize_text_field ( $params['slug'], '/' ) );
> On the other hand, it is arguable which variation is simpler.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/moxie-leean/wp-endpoints-author/pull/6/files/76f5f932261c27eeedca410cf9a5abd7f69ba717#r61332003>